test/functional/p2p_fingerprint.py

```diff
@@ -118,7 +118,7 @@ def run_test(self):
         block_hash = int(tip, 16)
         self.send_block_request(block_hash, node0)
         self.send_header_request(block_hash, node0)
-        node0.sync_with_ping()
+        node0.wait_for_block(block_hash, timeout=3)
```

Comment:
note for reviewers: it is correct usage of block_hash indeed (see bitcoin#20047).

Backport is full because remaining changes will be in bitcoin#20047 (otherwise code won't run correctly)